File: sticker/src/main/kotlin/chatox/sticker/messaging/rabbitmq/event/StickerPackUpdated.kt
package chatox.sticker.messaging.rabbitmq.event

import chatox.sticker.api.response.StickerPackResponse
import chatox.sticker.api.response.StickerResponse

data class StickerPackUpdated(
        val stickerPack: StickerPackResponse<*>,
        val newStickers: List<StickerResponse> = listOf(),
        val removedStickers: List<StickerResponse> = listOf()
)

File: uploads/src/external/types/StickerPackUpdated.ts
import {StickerPack} from "./StickerPack";
import {Sticker} from "./Sticker";

export interface StickerPackUpdated {
	stickerPack: Omit<StickerPack, "stickers">,
	newStickers: Sticker[],
	removedStickers: Sticker[]
}
